app/main.py

```diff
@@ -1,2 +1,57 @@
+import copy
+from typing import Union, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table = [[] for i in range(self.capacity)]
+
+    def __setitem__(self,
+                    key: Union[int, float, str, tuple, bool],
+                    value: Any
+                    ) -> None:
+        if self.length >= self.capacity * 2 / 3:
+            self.length = 0
+            self.capacity *= 2
+            old_table = copy.deepcopy(self.hash_table)
+            self.hash_table = [[] for i in range(self.capacity)]
+            for item in old_table:
+                if len(item) > 1:
+                    self.adding_item(item[0], item[2])
+
+        self.adding_item(key, value)
+
+    def adding_item(self,
+                    key: Union[int, float, str, tuple, bool],
+                    value: Any
+                    ) -> None:
+        hash_ = hash(key)
+        index_ = hash_ % self.capacity
```

Comment:
It's bad naming `hash_ index_,` use better `hash_data` or `hash_key` something like this